service/src/main/kotlin/fi/espoo/evaka/reservations/AttendanceReservationController.kt

```diff
@@ -403,30 +404,31 @@ private fun toChildDayRows(
             val child =
                 childData[childId]
                     ?: throw IllegalStateException("Child data not found for child $childId")
-            val hasMultipleRecordsOnSomeDay =
-                dailyData.any { placementStatus ->
-                    val date = placementStatus.date
-                    !placementStatus.inOtherUnit &&
-                        ((child.reservations[date] ?: listOf()).size > 1 ||
-                            (child.attendances[date] ?: listOf()).size > 1)
-                }
+            val maxRecordCountOnAnyDay =
+                max(
+                    1,
+                    dailyData.maxOf { placementStatus ->
+                        val date = placementStatus.date
+                        if (!placementStatus.inOtherUnit) {
+                            max(
+                                (child.reservations[date] ?: listOf()).size,
+                                (child.attendances[date] ?: listOf()).size
+                            )
+                        } else {
+                            0
+                        }
+                    }
+                )
+
             UnitAttendanceReservations.ChildDailyRecords(
                 child = child.child,
                 dailyData =
-                    listOfNotNull(
+                    (0 until maxRecordCountOnAnyDay).mapNotNull { rowIndex ->
```

Comment:
`associateBy` doesn't return `null`, so this should probably be just `map`?

Reply:
I think `0..maxRecordCountOnAnyDay` would be more idomatic Kotlin than using `until`.

Reply:
`0..maxRecordCountOnAnyDay` is inclusive, while `0 until maxRecordCountOnAnyDay` is exclusive. I find using `until` to be cleaner in this case.